tests/foreman/api/test_docker.py

```diff
@@ -39,49 +39,69 @@
 )
 from robottelo.test import APITestCase
 from robottelo.vm import VirtualMachine
+import string
 from time import sleep
 
+
 DOCKER_PROVIDER = 'Docker'
 
 
+def generate_alphanumeric_string_with_uppercase(length=10):
```

Comment:
your implementation looks really nice, but isn't easier to:
`''.join([gen_string('alphanumeric', length-1).lower(), gen_string('alpha', 1).upper()])`

your method makes sense when string is not random as for me

Reply:
We want to have string with upper character at random position.
In your case we will always have upper char in last position.